packages/data-stores/src/site/actions.ts

```diff
@@ -221,22 +221,30 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
 	function* setGlobalStyles(
 		siteIdOrSlug: number | string,
 		stylesheet: string,
-		globalStyles: GlobalStyles
+		globalStyles: GlobalStyles,
+		activatedTheme?: ActiveTheme
 	) {
-		const globalStylesId: number = yield getGlobalStylesId( siteIdOrSlug, stylesheet );
-
-		const updatedGlobalStyles: GlobalStyles = yield wpcomRequest( {
-			path: `/sites/${ encodeURIComponent( siteIdOrSlug ) }/global-styles/${ globalStylesId }`,
-			apiNamespace: 'wp/v2',
-			method: 'POST',
-			body: {
-				id: globalStylesId,
-				settings: globalStyles.settings ?? {},
-				styles: globalStyles.styles ?? {},
-			},
-		} );
+		// only update if there settings or styles to update
+		if (
+			Object.keys( globalStyles.settings ?? {} ).length ||
+			Object.keys( globalStyles.styles ?? {} ).length
+		) {
+			const globalStylesId: number =
+				activatedTheme?.global_styles_id || ( yield getGlobalStylesId( siteIdOrSlug, stylesheet ) );
+
+			const updatedGlobalStyles: GlobalStyles = yield wpcomRequest( {
+				path: `/sites/${ encodeURIComponent( siteIdOrSlug ) }/global-styles/${ globalStylesId }`,
+				apiNamespace: 'wp/v2',
+				method: 'POST',
+				body: {
+					id: globalStylesId,
+					settings: globalStyles.settings ?? {},
+					styles: globalStyles.styles ?? {},
```

Comment:
Default values are still needed because the if above is has an `||` one of these could be null.